modules/apps/transfer/keeper/relay_test.go

```diff
@@ -913,3 +913,87 @@ func (suite *KeeperTestSuite) TestOnTimeoutPacketSetsTotalEscrowAmountForSourceI
 	totalEscrowChainB = suite.chainB.GetSimApp().TransferKeeper.GetTotalEscrowForDenom(suite.chainB.GetContext(), coin.GetDenom())
 	suite.Require().Equal(sdkmath.ZeroInt(), totalEscrowChainB.Amount)
 }
+
+func (suite *KeeperTestSuite) TestPacketBackwardsCompatibility() {
+	// We are testing a scenario where a packet in the future has a new populated
+	// field called "new_field". And this packet is being sent to this module which
+	// doesn't have this field in the packet data. The module should be able to handle
+	// this packet without any issues.
```

Comment:
this test will test is effectively testing compatibility of new fields in the packet data in version N and version N+1 right? If we were to run an E2E and add a new field to packet data and execute a MsgTransfer against an older version of ibc-go (pre-json marshalling) this could still blow up right?

Reply:
actually I guess it just means that whichever version we are about to tag will be able to successfully process unknown fields, it doesn't necessarily mean it will be backwards compatible with a previous release right?

Reply:
Yes. Unless we backport this feature

Reply:
Maybe this should be called a future proof test?